### DISCUSSION TITLE
Deploying AES stack along with Nginx Ingress Controllers using this tutorial

### DISCUSSION THREAD

**Discussion by sonamsamdupkhangsar** [digitalocean/Kubernetes-Starter-Kit-Developers#131] [Q&A] [answered]
Hello,

I have already deployed the Nginx Ingress controller using the your published [Nginx Ingress tutorial](https://github.com/digitalocean/Kubernetes-Starter-Kit-Developers/blob/main/03-setup-ingress-controller/nginx.md).  I have installed the cert-manager and the issuer for the wildcard certificates for subdomain. 

I have a Helm chart that installs my deployment with the `ingressClassName: nginx`.  It is all working with Nginx ingress.

Now, I want to experiment with Ambassador Edge Stack (AES).  Can I install AES along with my Nginx ingress controller?  It seems like I should be able to as long as I have marked my previous deployments using Nginx Ingress with `ingressClassName: nginx`.  

Another question I had was regarding creating a new Certificate that will use my existing domain from my Nginx ingress setup by issuing this command: `kubectl apply -f 03-setup-ingress-controller/assets/manifests/ambassador/cert-manager-wcard-certificate.yaml`.  

Do you see any issues with the cert-manager on this for creating a new Certificate when I have already used that domain on my Nginx Ingress Wildcard certificate? 

I could create a new k8 cluster but I want to experiment with it on the same cluster if I could.

Thanks for any input.

-Sonam

**Comment by vladciobancai** [comment, marked as answer]
Hi @sonamsamdupkhangsar yes it is possible to deploy Nginx Ingress Controller and Ambassador Ingress Controller at the same time but you can not use the domain to point to both ingress controllers at the same time.

For certificate there will be no issue, you can request a new one and to be used by Ambassador . Please keep in mind that let's encrypt have a limitation of 5 certificate request per week, more details https://letsencrypt.org/docs/rate-limits/ For experimenting the certificate part / for testing we recomand to be used the let's encrypt staging environment https://letsencrypt.org/docs/staging-environment/ and reduce the chance of your running up against rate limits.

* for Ambassador you need to change the `server` to use the staging one https://github.com/digitalocean/Kubernetes-Starter-Kit-Developers/blob/main/03-setup-ingress-controller/assets/manifests/ambassador/cert-manager-wcard-issuer.yaml#L14 
* for Nginx you need to change the `server` to use the staging one https://github.com/digitalocean/Kubernetes-Starter-Kit-Developers/blob/main/03-setup-ingress-controller/assets/manifests/nginx/cert-manager-wcard-issuer.yaml#L14

**Reply by sonamsamdupkhangsar** [reply]
Thanks for the guidance.  

With your first line "yes it is possible to deploy Nginx Ingress Controller and Ambassador Ingress Controller at the same time but you can not use the domain to point to both ingress controllers at the same time."

I want to use a new subdomain like aes-test.mydomain.com for one app that will use AES but still leave my existing  nginxisubdomain.mydomain.com that uses the Nginx Ingress controller.  Just clarifying if I can still do that.

**Reply by vladciobancai** [reply]
Hi, Yes it is possible, in this case there will be no conflicts between both ingress controllers.
